File: CS/ViewModels/ContactViewModel.cs
namespace DXHtmlMessengerSample.ViewModels {
    using DevExpress.DevAV.Chat.Model;
    using DevExpress.Mvvm;
    using DevExpress.Mvvm.POCO;

    public class ContactViewModel : UserInfoViewModel {
        protected internal override string ServiceKey {
            get { return "ContactInfoFlyout"; }
        }
        public static ContactViewModel Create(UserInfo contact) {
            return ViewModelSource.Create(() => new ContactViewModel(contact));
        }
        //
        protected ContactViewModel(UserInfo contact)
            : base(contact) {
        }
        public void MailTo() {
            var msgService = this.GetRequiredService<IMessageBoxService>();
            msgService.ShowMessage("Mail To: " + Email);
        }
        public void VideoCall() {
            var msgService = this.GetRequiredService<IMessageBoxService>();
            int index = Email.IndexOf('@');
            msgService.ShowMessage("Video Call: " + Email.Substring(0, index));
        }
        public void PhoneCall() {
            var msgService = this.GetRequiredService<IMessageBoxService>();
            msgService.ShowMessage("Phone Call: " + MobilePhone);
        }
        public void TextMessage() {
            CloseDocument();
            Messenger.Default.Send(new Contact(Id, Name, Photo));
        }
    }
}

File: VB/ViewModels/ContactViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm
Imports DevExpress.DevAV.Chat.Model

Namespace DXHtmlMessengerSample.ViewModels

    Public Class ContactViewModel
        Inherits UserInfoViewModel

        Protected Friend Overrides ReadOnly Property ServiceKey() As String
            Get
                Return "ContactInfoFlyout"
            End Get
        End Property
        Public Shared Function Create(ByVal contact As UserInfo) As ContactViewModel
            Return ViewModelSource.Create(Function() New ContactViewModel(contact))
        End Function
        '
        Protected Sub New(ByVal contact As UserInfo)
            MyBase.New(contact)
        End Sub
        Public Sub MailTo()
            Dim msgService = Me.GetRequiredService(Of IMessageBoxService)()
            msgService.ShowMessage("Mail To: " & Email)
        End Sub
        Public Sub VideoCall()
            Dim msgService = Me.GetRequiredService(Of IMessageBoxService)()
            Dim index As Integer = Email.IndexOf("@"c)
            msgService.ShowMessage("Video Call: " & Email.Substring(0, index))
        End Sub
        Public Sub PhoneCall()
            Dim msgService = Me.GetRequiredService(Of IMessageBoxService)()
            msgService.ShowMessage("Phone Call: " & MobilePhone)
        End Sub
        Public Sub TextMessage()
            CloseDocument()
            Messenger.Default.Send(New Contact(Id, Name, Photo))
        End Sub
    End Class
End Namespace